src/replay_stage.rs

```diff
@@ -449,6 +433,40 @@ impl ReplayStage {
         let next_slots = blocktree.get_slots_since(&[slot_index]).expect("Db error");
         next_slots.first().cloned()
     }
+
+    fn forward_entries(
+        entries: &[Entry],
```

Comment:
Since this function consumes the entries, this should be `Vec<Entry>`, not `&[Entry]`.

Reply:
It currently doesn't consume the entries, since they need to exist for ReplayStage::process_entries

Reply:
The entries are consumed here: https://github.com/solana-labs/solana/pull/2985/files#diff-3d5b06a60c5839fe958a3e8438bce845R467

The one clone that's needed is on the `entries` parameter when you call this function.